Updated code to GNU Radio 3.10

dkozel opened the pull request:
Hello. I just had a chance to quickly test this on an SMU-200 and it worked pretty well. I updated it to GNU Radio 3.10 and fixed one small bug where the "CLOCK" property in the file was not being set to the sample rate.

There is one decrease in functionality, GNU Radio changed it's audio sink and the function to automatically handle converting from host endianness to little-endian was removed. I brute force copied that code from GNU Radio 3.7 but didn't keep the check for host endianness so the block currently won't work if the host is big endian.

If you're interested in merging this update I can fix that and restore all the copyright notices correctly.

![rs-waveforms-demo](https://github.com/AndersKaloer/gr-rswaveforms/assets/253888/c598a18d-9614-43fa-aaa0-4fc61a282801)

Commit:
Updated code to GNU Radio 3.10

AndersKaloer commented:
Excellent, thanks a lot!
It would be great if you could fix the endianness issue, although I am not sure how important it is in practice. I guess it also would be fine to just support little endian hosts.

I would be happy to test on hardware, but I will not have access until after the holidays.

Commit:
Readded author information, added basic docs

Commit:
Added and ran clang-format

Used the .clang-format file from GNU Radio 3.10.9.1

Commit:
Re-added cosine example

dkozel commented:
I don't have a big-endian host to test with, but I'm reasonably confident that this solution handles the sample conversion if needed. Once C++20 becomes required for GNU Radio there's some additional simplification possible, though the first step really is to interleave the IQ into one array in memory, byte swap all the elements in one pass, and write the entire buffer to the file to allow the compiler more room for optimisation.

I don't have any hardware to test with, but the R&S ARB Toolkit reads the new file correctly.
![image](https://github.com/AndersKaloer/gr-rswaveforms/assets/253888/aaeb044c-3c57-497f-b4e6-cfbdc213ecad)

Commit:
Re-added check for native endianness

GNU Radio's CMake provides a flag to check for a big-endian host. A
templated ByteSwap function was added to convert to little-endian if
needed.

dkozel commented:
Commits are tidied, if you would like to to squash all these into one I'm happy to do that. Thanks for writing the module!

nickoe commented:
Hi @dkozel

Thank you very much for updating this, it is very helpful and much appreciated.  I was wondering if there is a chance to upstream this to gnuradio if there is no licensing issues, I guess this would make it more useful over time and maybe someone would implement a source for end to end testing?

I just tried to build this on archlinux on my workstation as is, but I get a build error. 

I wonder why I see this in the cmake configure:
```
...
-- Could NOT find PythonLibs: Found unsuitable version "2.7.18", but required is exact version "3.11" (found /usr/lib/libpython2.7.so)
-- Extracting version information from git describe...
fatal: No names found, cannot describe anything.
...
```

I do have 3.11.. and not sure why git describe also fails.  I guess those are minor issues.  I can't easily test it now and R&S ARB Toolbox does not run for me in wine.

EDIT2: I updated my comment to reflect reality :D

nickoe commented on MANIFEST.md:
Maybe keep the old info from the manifest md?

nickoe commented on MANIFEST.md:
```diff
@@ -1,10 +1,17 @@
-
+  - Copyright Owner 1
+license:
+gr_supported_version: # Put a comma separated list of supported GR versions here
```
3.10, 3.11?

nickoe commented on README.md:
```diff
@@ -1,17 +1,17 @@
 # gr-rswaveforms
-GNURadio sink that creates Rohde & Schwarz SMU waveform files (.wv) for R&S signal generators (e.g. SMU, SMBV).
+
+A GNU Radio module for creating Rohde & Schwarz SMU waveform files (.wv) for Rohde & Schwarz signal generators (e.g. SMU, SMV).
```
I am not familiar with the SMV, but I know the SMBV so I don't think it was a typo per se.

dkozel replied:
Changed SMV back to SMBV. :+1:

dkozel commented:
@nickoe Thanks for pointing out the Manifest file. I'd completely overlooked it. I've added that back in and squashed it into 00cfca3.

Implementing a source block seems quite straightforward and would be a nice addition, I agree.

I don't think this would be considered for upstreaming to the core, if anything the core is likely to be trimmed down significantly in the 4.0 release.  I am working with Kitware to significantly streamline the installation and management of OOT modules, so hopefully these become easier for people to find and use. There's also hopes for CI for all these OOT modules so authors can get feedback about when changes to GR or operating systems break things.

AndersKaloer commented:
Thank you both, @dkozel and @nickoe, for implementing this!
I trust the R&S ARB Toolkit for now, but hope to be able to verify it on HW soon.
